File: android/src/main/java/com/capacitortoastwithduration/ToastWithDurationPlugin.java
package com.capacitortoastwithduration;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "ToastWithDuration")
public class ToastWithDurationPlugin extends Plugin {

    private ToastWithDuration implementation = new ToastWithDuration();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

      @PluginMethod
    public void show(PluginCall call) {
        String text = call.getString("text");
        if (text == null) {
            call.reject("Must provide text");
            return;
        }

        String durationType = call.getString("duration", "short");

        int duration = android.widget.Toast.LENGTH_SHORT;
        if ("long".equals(durationType)) {
            duration = android.widget.Toast.LENGTH_LONG;
        }
        String position = call.getString("position", "bottom");
        ToastWithDuration.show(getContext(), text, duration, position);

        call.resolve();
    }
}

File: ios/Plugin/ToastWithDurationPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ToastWithDurationPlugin)
public class ToastWithDurationPlugin: CAPPlugin {
    private let implementation = ToastWithDuration()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

     @objc func show(_ call: CAPPluginCall) {
        guard let text = call.getString("text") else {
            call.reject("text must be provided and must be a string.")
            return
        }
        var duration = call.getInt("durationMilliseconds", 0)
        if(duration <= 0) {
            let durationType = call.getString("duration", "short")
            duration = durationType == "long" ? 3500 : 2000
        }
        let position = call.getString("position", "bottom")

        guard let viewController = bridge?.viewController else {
            call.reject("Unable to display toast!")
            return
        }
        implementation.showToast(in: viewController, text: text, duration: duration, position: position, completion: {(_) in
            call.resolve()
        })
    }
}
